jetstream/stream.go

```diff
@@ -259,28 +260,52 @@ func (s *stream) Info(ctx context.Context, opts ...StreamInfoOpt) (*StreamInfo,
 	}
 	var req []byte
 	var err error
-	if infoReq != nil {
-		req, err = json.Marshal(infoReq)
-		if err != nil {
-			return nil, err
-		}
-	}
+	var subjectMap map[string]uint64
+	var offset int
 
 	infoSubject := apiSubj(s.jetStream.apiPrefix, fmt.Sprintf(apiStreamInfoT, s.name))
-	var resp streamInfoResponse
-
-	if _, err = s.jetStream.apiRequestJSON(ctx, infoSubject, &resp, req); err != nil {
-		return nil, err
-	}
-	if resp.Error != nil {
-		if resp.Error.ErrorCode == JSErrCodeConsumerNotFound {
-			return nil, ErrStreamNotFound
+	for {
+		if infoReq != nil {
+			if infoReq.SubjectFilter != "" {
+				if subjectMap == nil {
+					subjectMap = make(map[string]uint64)
+				}
+				infoReq.Offset = offset
+			}
+			req, err = json.Marshal(infoReq)
+			if err != nil {
+				return nil, err
+			}
+		}
+		var resp streamInfoResponse
+		if _, err = s.jetStream.apiRequestJSON(ctx, infoSubject, &resp, req); err != nil {
+			return nil, err
+		}
+		if resp.Error != nil {
+			if resp.Error.ErrorCode == JSErrCodeStreamNotFound {
+				return nil, ErrStreamNotFound
+			}
+			return nil, resp.Error
+		}
+		s.info = resp.StreamInfo
+		var total int
+		if resp.Total != 0 {
+			total = resp.Total
+		}
+		if len(resp.StreamInfo.State.Subjects) > 0 {
+			for subj, msgs := range resp.StreamInfo.State.Subjects {
+				subjectMap[subj] = msgs
+			}
+			offset = len(subjectMap)
+		}
+		if total == 0 || total <= offset {
+			resp.StreamInfo.State.Subjects = subjectMap
```

Comment:
I wonder if we should store all the subjects in the cached info, as this might really grow with time.

Reply:
yeah, I agree, that might be too much, I'll leave the cached info without the subjects

Reply:
possibly if they asked for it, you can save it on the cache, but if the next info doesn't require that, simply replace

Reply:
I removed caching subjects map